psychopy/visual/window.py

```diff
@@ -383,6 +383,13 @@ def __del__(self):
         if self._closed!=False:
             self.close()
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        if self._closed != False:
```

Comment:
Good spot. I was actually just copying `__del__`, which I was assuming would need a similar cleanup to `__exit__`. So I think that probably needs fixing, too. Is that okay to put in the same PR?